geoips/interfaces/base.py

```diff
@@ -460,8 +412,59 @@ def get_plugin(self, name):
         ProductsInterface which uses a tuple containing 'source_name' and
         'name'.
         """
+        if not self._unvalidated_plugins:
+            raise PluginRegistryError(
+                "Plugin registry not found, please run 'create_plugin_registry'"
+            )
         try:
-            validated = self.validator.validate(self._unvalidated_plugins[name])
+            if isinstance(name, tuple):
+                # These are stored in the yaml as str(name),
+                # ie "('viirs', 'Infrared')"
+                plugin = yaml.safe_load(
+                    open(
+                        self._unvalidated_plugins[self.name][name[0]][name[1]][
+                            "abspath"
+                        ],
+                        "r",
+                    )
+                )
+                plugin_found = False
+                for product in plugin["spec"]["products"]:
+                    if (
+                        product["name"] == name[1]
+                        and name[0] in product["source_names"]
+                    ):
+                        plugin_found = True
+                        plugin = product
+                        break
+                if not plugin_found:
+                    raise PluginError(
+                        "There is no plugin that has " + name[1] + " included in it."
+                    )
+                plugin["interface"] = "products"
+                plugin["abspath"] = self._unvalidated_plugins[self.name][name[0]][
+                    name[1]
+                ]["abspath"]
+                plugin["relpath"] = self._unvalidated_plugins[self.name][name[0]][
+                    name[1]
+                ]["relpath"]
+                plugin["package"] = self._unvalidated_plugins[self.name][name[0]][
+                    name[1]
+                ]["package"]
```

Comment:
Can this section be simplified by setting unval_plugin = self._unvalidated_plugins[self.name][name[0]][name[1]] then using that?

Reply:
Not to my knowledge. We have to load in the appropriate plugin, then add the appropriate information (package, relpath, abspath), to that yaml dictionary. If you can think of another method though, I'm all ears. This relates to #383.

Reply:
Would it work to just set it to self._unvalidated_plugins[self.name][name[0]][name[1]], then just update that dictionary with the full contents of the plugin, using merge_nested_dict?

Reply:
This is fine as is and will be moot when #389 is done.